cmd/weaver-kube/deploy.go

```diff
@@ -147,3 +152,42 @@ func deploy(ctx context.Context, args []string) error {
 	// Generate the kube deployment information.
 	return impl.GenerateKubeDeployment(image, dep, config)
 }
+
+// checkVersionCompatibility checks that the tool binary is compatible with
+// the application binary being deployed.
+func checkVersionCompatibility(appBinary string) error {
+	versions, err := bin.ReadVersions(appBinary)
+	if err != nil {
+		return fmt.Errorf("read versions: %w", err)
+	}
+	selfVersion, _, err := impl.ToolVersion()
+	if err != nil {
+		return fmt.Errorf("read weaver-kube version: %w", err)
+	}
+	if versions.DeployerVersion != version.DeployerVersion {
+		// Try to relativize the binary, defaulting to the absolute path if
+		// there are any errors..
+		binary := appBinary
+		if cwd, err := os.Getwd(); err == nil {
+			if rel, err := filepath.Rel(cwd, appBinary); err == nil {
+				binary = rel
+			}
+		}
+		return fmt.Errorf(`
+	ERROR: The binary you're trying to deploy (%q) was built with
+	github.com/ServiceWeaver/weaver module version %s. However, the 'weaver-kube'
+	binary you're using was built with weaver module version %s. These versions are
+	incompatible.
+	
+	We recommend updating both the weaver module your application is built with and
+	updating the 'weaver-kube' command by running the following.
+	
+		go get github.com/ServiceWeaver/weaver@latest
```

Comment:
is this still needed? Same comment for the next line.

Reply:
I think it's still needed. If we detect a version mismatch, we recommend the user updates to the latest version of the weaver module and the latest version of the deployer to make sure they are compatible.

Reply:
Yeah, this is detecting the deployer protocol version compatibility, which is still an issue.